astropy/io/fits/hdu/table.py

```diff
@@ -295,6 +298,13 @@ def __init__(self, data=None, header=None, name=None, uint=False):
                 # Make a "copy" (not just a view) of the input header, since it
                 # may get modified.  the data is still a "view" (for now)
                 hcopy = header.copy(strip=True)
+
+                # Separate global and column-specific keywords
+                for key in hcopy.keys():
+                    if re.search(r'\d+$', key):
```

Comment:
Just because a keyword ends with digits doesn't mean it must be a column key, but that seems to be implied here and below.